code/data_setup/parse_imagenet_bb.py

```diff
@@ -7,15 +7,47 @@
 
 def parse_file(bbox_dir, filepath): 
     tree = ET.parse(bbox_dir + filepath)
-    print root.findall('./')
-    pass
+    root = tree.getroot()
 
+    output_lst = []
+    size_node = root.find('size')
+    width = size_node.find('width').text
+    height = size_node.find('height').text
+
+    for child in root.findall('./'): 
+        if child.tag == "object": 
+            object_lst = []
+            object_lst.append(width)
+            object_lst.append(height)
+            name = child.find('name').text
+            subcategory = child.find('subcategory')
+            subcategory = None if subcategory is None else subcategory.text
+
+            bounding_box = child.find('bndbox')
+            xmin = bounding_box.find('xmin').text
+            xmax = bounding_box.find('xmax').text
+            ymin = bounding_box.find('ymin').text
+            ymax = bounding_box.find('ymax').text
+
+            object_lst.append(name)
+            object_lst.append(subcategory)
+            object_lst.append(xmin)
+            object_lst.append(xmax)
+            object_lst.append(ymin)
+            object_lst.append(ymax)
+            output_lst.append(object_lst)
+
```

Comment:
What about replacing all the individual append commands in this function with
`
object_lst.extend([width, height, name, subcategory, xmin, xmax, ymin, ymax, object_lst])
`